lib/ecto/query/builder.ex

```diff
@@ -555,23 +559,23 @@ defmodule Ecto.Query.Builder do
   @doc """
   Validates the type with the given vars.
   """
-  def validate_type!({composite, type}, vars) do
-    {composite, validate_type!(type, vars)}
+  def validate_type!({composite, type}, vars, env) do
+    {composite, validate_type!(type, vars, env)}
   end
-  def validate_type!({:^, _, [type]}, _vars),
-    do: type
-  def validate_type!({:__aliases__, _, _} = type, _vars),
+  def validate_type!({:^, _, [type]}, _vars, _env),
     do: type
-  def validate_type!(type, _vars) when is_atom(type),
+  def validate_type!({:__aliases__, _, _} = type, _vars, {env, _}),
+    do: Macro.expand(type, env)
```

Comment:
Why do we need to expand the alias? It is fine to return it as an alias as it will be expanded by the compiler.

Reply:
Without expanding the alias it didn't work trying to call `.type()` on the AST instead of on the module.

Reply:
Ah, I see. Good call then!